airflow/models.py

```diff
@@ -2485,7 +2485,10 @@ def __lt__(self, other):
     def __hash__(self):
         hash_components = [type(self)]
         for c in self._comps:
-            val = getattr(self, c, None)
+            if c == 'task_ids':
```

Comment:
Ah, in that case you could use a frozenset or tuple instead of a list for the existing task_ids property (immutability wherever possible is generally a good thing anyways), or define a new property like task_ids_as_tuple. I am OK with the current solution though, this is definitely a nit.